app/controllers/users/registrations_controller.rb

```diff
@@ -25,14 +25,21 @@ def new
   def create
     if valid_invitation_code?
       @user = User.new(invited_user_params)
-      @user.roles << invitation(session[:invitation_code]).role
+
+      invitation = Invitation.find_by(invitation_code: session[:invitation_code])
+      if invitation.enroll_elligible?
+        @user.roles << Role.find_by(name: "enroll-elligible")
+      else
+        @user.roles << invitation.role
+      end
```

Comment:
I'm wondering why we set "enroll-elligible" here but other roles get assigned directly. I see in invitation.rb and invitation_manager.rb we are setting the role as "invitee" but then here we set the role as "enroll-elligible" after checking that it has a role of "invitee". Do we need this role to be different at different times or can the dropdown say "Invitee" but then they get assigned a role of "enroll-elligible"?

Reply:
Yup, good question. I thought about this too. A couple things:

1) this type of invitation is designed to be used by only the API so it shouldn't be in the dropdown but your question is still a good one!

2) we need to end up with a role of `enroll-elligible` at the end so that that user can log into Enroll. We could skip the whole `invitee` role idea as you said but I like having the API accept `invitee` since that word is what needs show up in Apply (when it calls this API). The fact that under the hood we apply the `enroll-elligble` role is irrelevant to Apply, so it's a nice separation of concerns but it does add this little indirection you called out.

Lemme see if I can come up with a more compelling reason to keep it...

Reply:
This might not be a great solution since you'd have to touch all three applications to implement it, but if we call it `enrollable` then that role makes sense for Apply, Census, and Enroll. 

Unless there is a nuance between `invitee` meaning they are invited but they don't get access to Census or Enroll until they accept and can therefore enroll/be `enroll-elligible`. But from what I currently understand that is not the case. Maybe I don't have it all straight though.

Reply:
`enrollable` to me sounds like they are able to enroll. `enroll-elligible` is more than that in that it allows them to enroll and then access Enroll indefinitely (to make payments for example).

Let's try removing the `invitee` idea and going with just `enroll-elligible` on both the invitation and the user role. Good question!

Reply:
Done in ad482f9